tests/test_monitor.py

```diff
@@ -0,0 +1,77 @@
+# All tests pass successfully individually but not as a suite
+import luigi
+from luigi_monitor import monitor
+import unittest
+import os
+import mock
+import requests
+import inspect
+
+
+class TestMissingTask(luigi.ExternalTask):
+
+    def output(self):
+        return luigi.LocalTarget('dummy.txt')
+
+
+class TestSuccessTask(luigi.Task):
+    num = luigi.Parameter()
+
+    def output(self):
+        return luigi.LocalTarget('words.txt')
+
+    def run(self):
+        # write a dummy list of words to output file
+        words = [
+            'apple',
+            'banana',
+            'grapefruit'
+        ]
+
+        with self.output().open('w') as f:
+            for word in words:
+                f.write('{word}\n'.format(word=word))
+
+
+# This method will be used by the mock to replace requests.get
+def mocked_requests_post(*args, **kwargs):
+    resp = mock.Mock(spec=requests.Response)
+    resp.status_code = 200
+    return resp
+
+
+class TestLuigiMonitor(unittest.TestCase):
+    @mock.patch('requests.post', side_effect=mocked_requests_post)
+    def test_monitor_no_message(self, mock_post):
+        with monitor(slack_url='mock://slack', events=['DEPENDENCY_MISSING']) as m:
+            luigi.run(main_task_cls=TestSuccessTask, local_scheduler=True, cmdline_args=['--num', '1'])
+            self.assertDictEqual(m.recorded_events, {})
+        self.assertEqual(mock_post.call_count, 0, 'Slack webhook called')
+
+    @mock.patch('requests.post', side_effect=mocked_requests_post)
```

Comment:
I've always used `new=mocked_requests_post` here. `mock`'s docs aren't really helping me, does `side_effect` do the same thing?

Reply:
`side_effect` allows defining a function call to replace the mocked call. should have same signature.